### PR TITLE
Hold every test this package names in its prose to existing

### LINKED ISSUE #258: Nothing checks that a test named in prose is still a test
#256 is about two landmarks in a module docstring that pointed at the
wrong test, and its fix names the tests instead of their position. That is
strictly better — an append cannot falsify a name — but it is not
*checked*: a rename falsifies it just as silently, and nothing here would
go red.

That is not hypothetical. #255's rename moved two test names, and the only
reason no prose went stale with them is that a human grepped for
citations; the changelog line that cites one of them was updated by hand in
that same diff. Test names in this suite are prose that other prose refers
to — the docstring of `tests/test_verified_signing.py` now names four of
them — so the class is growing rather than shrinking.

**What exists to check.** Every citation of the form `` `test_...` `` in
`tests/*.py`, which today is eleven of them across six modules:
`test_the_sweep_is_whole`, `test_pubkey_from_prvkey`,
`test_the_sender_wipes_the_keys_it_had_when_a_later_one_is_refused`,
`test_every_private_half_is_paired`,
`test_the_outputs_follow_the_recipient_order`, and the five in
`tests/test_verified_signing.py`. All eleven resolve to a test that exists
today — measured on d9f14de, so this issue is a guard against a defect
rather than a report of one.

**And the wrinkle that makes it a design question rather than three
lines.** One citation is not ours: `tests/test_vectors.py:36` and `:1154`
name `test_low_r`, which is rust-secp256k1's test in its own `src/lib.rs`,
cited to say which upstream vector is reproduced — correct prose that a
naive walk would report as a dangling reference. So the check has to tell a
citation of a local test from a citation of an upstream one, and the
options are not equal: matching only names that a test module defines
somewhere would silently pass a genuine typo, while excluding a list of
upstream names by hand is a list nothing maintains.

Where such a walk lives is the other half of the question. `tests/test_secret.py`
already walks the package for callers of `_secret.take` and is the
precedent for a test that reads the tree; a check over the *test* modules'
prose is a different population and may not belong beside it.

Noticed while fixing #256 and not addressed there: that diff is three
sentences of one docstring, and this is a check over every module's prose.

### PR DESCRIPTION
**Pull request by fametrano**
[PR 260](https://github.com/btclib-org/btclib-secp256k1/pull/260) replaced
two positional landmarks with the names of the tests they meant, and said
what a name does not buy:
[PR 257](https://github.com/btclib-org/btclib-secp256k1/pull/257) had
renamed two of the four tests that docstring cites, and nothing in the suite
would have gone red. `tests/test_citations.py` is that check.

It reads every backticked span in this suite **and in the package's own
sources**, and holds each one that spells a test name to being a test that
exists. The package half is not incidental: `btclib_secp256k1/dsa.py` names
two of the cases behind a claim it makes about itself and
`btclib_secp256k1/recovery.py` names one, so a rename can falsify a shipped
docstring.

**The reader is the design, and a one-line pattern has two shapes wrong.**

- **A citation is wrappable.** `tests/test_docs.py` cites
  `test_no_documented_module_has_gone_away` broken at an underscore across
  two lines. A pattern anchored to one line does not report that as
  dangling — it never sees it, which is the failure a guard must not have.
  So a span is collapsed before it is compared, of whitespace and of the
  `#` a wrap inside a comment block carries into the middle of the name.
  Found by the reviewer trying the naive version, and recorded on
  [ISS 258](https://github.com/btclib-org/btclib-secp256k1/issues/258).
- **A cited name may be somebody else's.** `tests/test_vectors.py` names
  rust-secp256k1's own `test_low_r` to say which upstream vector it
  reproduces — correct prose resolving to nothing here. Those are listed
  with where each comes from, and each is held to *still being cited*, so
  the list cannot become names nobody removed.

Text rather than the syntax tree, which inverts `tests/test_secret.py`'s
reasoning for the opposite population: there a mention had to be told from
a call, so the names come off the AST; here the mention is the subject, and
half of them are in comments no tree carries.

**What the module does not read is itself**, and that is what makes the
staleness half mean anything: it names the exempted test in order to explain
the exemption, so reading itself would let the guard keep its own exemptions
alive. Its reader's cases are citations held as fixtures rather than prose,
one of them a name that deliberately does not exist — which passed by luck
before it was a decision, the literals' `\n` being an escape rather than a
line break.

`CHANGELOG.md` and `HISTORY.md` are out of scope, and not for the twenty or
so citations they would add: a released section is that release's account of
itself, so a check over it would ask for the record to be edited whenever
the tree moves. Both names there that resolve to nothing are foreign anyway
— Core's `test_case` and rust-secp256k1's `test_low_r`.

Evidence, local, on `f08f61c` — **four breakages by hand**, each restored,
`PYTHONDONTWRITEBYTECODE=1` throughout:

- the wrapped citation's definition renamed and nothing else —
  `test_every_cited_test_exists` fails, which is the case a single-line
  pattern misses
- a definition renamed that only the *package* cites,
  `test_der_reaches_all_72_octets` — fails
- both `test_low_r` citations removed —
  `test_no_foreign_citation_has_gone_stale` fails
- `_SPAN` broken so the reader finds nothing — both walks pass vacuously
  and `test_the_prose_was_read_at_all` fails, which is why it is there

and the gates:

- `uv run --locked --no-default-groups --group test pytest --cov` — exit 0,
  734 passed, 100.00%
- `uvx pre-commit run --all-files` — exit 0, 37 hooks, nothing modified
- `uv build --sdist` and `twine check --strict` — exit 0 both, the new
  module present in the sdist

Closes #258

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Prevent test-name references in source and test prose from silently becoming stale.

Bug Fixes:
- Add coverage that detects stale or missing test-name citations in package and test prose, including wrapped citations and foreign test references.

Enhancements:
- Validate citation parsing independently of test definitions and ensure the citation reader cannot pass vacuously.
- Track foreign test citations so exemptions remain valid only while those references are still used.

Documentation:
- Document the new safeguards and the scope of source files checked for test-name citations.

Tests:
- Add citation-reader tests covering multiline comments, non-citation code spans, local and foreign test names, and empty-reader failures.

### CODE REVIEW

**Comment by fametrano** [comment]
@claude

File: tests/test_citations.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# Copyright (c) The btclib developers
+#
+# Distributed under the MIT software license, see the accompanying
+# LICENSE file or https://opensource.org/license/mit for the full text.
+
+"""A test this package names in its prose is a test that exists.
+
+The prose here points at tests by name -- the module docstring of
+`tests/test_verified_signing.py` names four, and `btclib_secp256k1/dsa.py`
+names two of the cases behind a claim it makes about itself -- and nothing
```
[the remark this answers is not in the document]
Comment (fametrano) [reply tests/test_citations.py]: Fixed in 1490753: reworded to drop the count entirely ("names several") rather than correcting it to "five", since this module's whole purpose is exactly the staleness this sentence itself fell into. Also brought the branch up to date with main (#267, #268, #270-272).

File: tests/test_citations.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+- **a name that is not ours.** `tests/test_vectors.py` cites
+  rust-secp256k1's own `test_low_r` to say which upstream vector it
+  reproduces: correct prose that resolves to nothing here. Those are named
+  in `_FOREIGN`, with where each comes from, and each is asserted to still
+  be cited by something other than this file -- an exemption that stops
+  applying fails rather than sitting there, which it could not do if the
+  prose explaining it counted as a use.
+
+`CHANGELOG.md` and `HISTORY.md` are out of scope, and not for the twenty
+or so citations they would add. A released section is that release's
```
[the remark this answers is not in the document]
Comment (fametrano) [reply tests/test_citations.py]: Fixed in ac61623: dropped the count, keeping only the reasoning it added nothing to ("not for the citations they would add").

[the remark this answers is not in the document]
Comment (fametrano) [reply tests/test_citations.py]: Fixed in 37b900a: named cases without counting them, in both places (this docstring and its CHANGELOG.md entry).

File: CHANGELOG.md [hunk 3]
```diff
@@ -608,6 +608,36 @@ release-notes length in the first place, and are still in
   still does not buy is a check -- a rename falsifies it as quietly as an
   append falsified a position -- which is #258, and the count in the same
   docstring is #259.
+- **And that check exists now** (#258). `tests/test_citations.py` reads
+  every backticked span in this suite and in the package's own sources,
+  and holds each one that spells a test name to being a test that exists.
+  The package half is not incidental: `btclib_secp256k1/dsa.py` names two
+  of the cases behind a claim it makes about itself and
+  `btclib_secp256k1/recovery.py` names one, so a rename can falsify a
```
[the remark this answers is not in the document]
Comment (fametrano) [reply CHANGELOG.md]: Fixed in 37b900a: named cases without counting them.

File: tests/test_citations.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
+# Copyright (c) The btclib developers
+#
+# Distributed under the MIT software license, see the accompanying
+# LICENSE file or https://opensource.org/license/mit for the full text.
+
+"""A test this package names in its prose is a test that exists.
+
+The prose here points at tests by name -- the module docstring of
+`tests/test_verified_signing.py` names several, and `btclib_secp256k1/dsa.py`
+names two of the cases behind a claim it makes about itself -- and nothing
```
[the remark this answers is not in the document]
Comment (fametrano) [reply tests/test_citations.py]: Fixed in 37b900a: named cases without counting them.

File: tests/test_citations.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+_SOURCES = [
+    path
+    for path in sorted((_ROOT / "tests").glob("*.py"))
+    + sorted((_ROOT / "btclib_secp256k1").glob("*.py"))
+    if path.name != Path(__file__).name
+]
+
+# any backticked span, across lines: what is inside is decided after the
+# whitespace comes out, a citation being wrappable at an underscore
+_SPAN = re.compile(r"`([^`]+)`", re.DOTALL)
```
[the remark this answers is not in the document]
Comment (fametrano) [reply tests/test_citations.py]: Fixed in ba18409: dropped the flag, and reworded the comment to say the negated character class is what crosses the line.
